platform/ios/MGLMapView.mm

```diff
@@ -1032,10 +1032,11 @@ - (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecogni
 
 - (void)trackGestureEvent:(NSString *)gesture forRecognizer:(UIGestureRecognizer *)recognizer
 {
+    CGPoint ptInView = CGPointMake([recognizer locationInView:recognizer.view].x, [recognizer locationInView:recognizer.view].y);
+    const CLLocationCoordinate2D coord = [self convertPoint:ptInView toCoordinateFromView:recognizer.view];
```

Comment:
> We may want to see if we can do this in a `const` method in a non-main thread.

Yeah, I meant more changing the C++ to make this safer somehow for non-main threads — like snapshotting the `TransformState` or something. Looking into it.